### PR TITLE
Avoid error calling addEventListener in browsers that don't support it

### PR DESCRIPTION
**Pull request by javan**
Fix for #173

### CODE REVIEW

File: src/turbolinks/index.coffee [hunk 1]
```diff
@@ -7,7 +7,9 @@
 
 @Turbolinks =
   supported: do ->
-    window.history.pushState? and window.requestAnimationFrame?
+    window.history.pushState? and
+      window.requestAnimationFrame? and
+      window.addEventListener?
```
Comment (javan) [inline src/turbolinks/index.coffee]: Not technically needed to fix the issue, but seems reasonable to add for clarity.

File: src/turbolinks/history.coffee [hunk 2]
```diff
@@ -35,11 +30,18 @@ class Turbolinks.History
...
+
   # Private
 
   shouldHandlePopState: ->
     # Safari dispatches a popstate event after window's load event, ignore it
-    pageLoaded is true
+    @pageIsLoaded()
+
+  pageIsLoaded: ->
+    @pageLoaded or document.readyState is "complete"
```
Comment (javan) [inline src/turbolinks/history.coffee]: Added an additional check for `document.readyState` to cover the theoretical case where turbolinks is loaded async after the `load` event.

Comment (nateberkopec) [inline src/turbolinks/history.coffee]: 👍 Similar to [what JQuery does](https://github.com/jquery/jquery/blob/master/src/core/ready.js#L80).

**Comment by printercu** [comment]
Is there plan to release this fix? I've found that `addEventListener("load", ...)` is not firing event sometimes in iOS simulator, so `history.back()` doesn't work. This patch seems to fix the issue.

**Pull request merged**